### PR TITLE
Add some conformance tests

### PR DESCRIPTION
**Pull request by fingolfin**
This is just a very basic start, as a proof of concept.
More rings should be tested, and more conformance tests be made.
Also, some existing tests may now be redundant and could be removed.

### CODE REVIEW

File: test/flint/fmpq-test.jl [hunk 1]
```diff
@@ -1,3 +1,13 @@
+function test_elem(R::FlintRationalField)
+   n = fmpz(2)^rand(1:100)
+   return fmpz(rand(-n:n)) // fmpz(rand(fmpz(1):n))
```
Comment (tthsqe12) [inline test/flint/fmpq-test.jl:3]: Could be the slightly simpler 
```julia
   return rand_bits(ZZ, rand(0:100))/rand_bits(ZZ, rand(1:100))
```

File: test/flint/fmpz-test.jl [hunk 2]
```diff
@@ -1,3 +1,12 @@
+function test_elem(R::FlintIntegerRing)
+   n = fmpz(2)^rand(1:100)
+   return fmpz(rand(-n:n))
```
Comment (tthsqe12) [inline test/flint/fmpz-test.jl:3]: Ditto
```julia
   return rand_bits(ZZ, rand(0:100))
```

File: test/flint/fq-test.jl [hunk 3]
```diff
@@ -1,3 +1,21 @@
+function test_elem(R::FqFiniteField)
+   n = rand(fmpz(0):order(R))
+   iszero(n) && return zero(R)
+   return gen(R)^n
```
Comment (tthsqe12) [inline test/flint/fq-test.jl:4]:
```suggestion
   return rand(R)
```
The previous code leads one to believe that `gen(R)` returns a generator for the multiplicative group, which it usually does but doesn't necessarily do (sorry :(). In order to not miss any elements, the `rand` function should work just fine.
We could also increase the probability of returning zero, which might test more corner cases...

File: test/flint/gfp-test.jl [hunk 4]
```diff
@@ -1,3 +1,14 @@
+function test_elem(R::Nemo.GaloisField)
+   return rand(R, fmpz(0):characteristic(R))
```
Comment (tthsqe12) [inline test/flint/gfp-test.jl:2]: `rand(R)` works here too :)

File: test/flint/padic-test.jl [hunk 5]
```diff
@@ -1,3 +1,17 @@
+function test_elem(R::FlintPadicField)
+   p = prime(R)
+   prec = rand(1:R.prec_max)
+   r = fmpz(0):p-1
+   return R(sum(rand(r)*p^i for i in 0:prec))
+end
+
+@testset "padic.conformance_tests" begin
+# TODO: make the following work; for now they fail because the conformance
+# tests want to use isapprox on padic elements, but no such method exists
```
Comment (tthsqe12) [inline test/flint/padic-test.jl:10]: Sorry about this. The conformance tests did not pass with BigFloat and ==. I think it is better to remove BigFloat as a conforming Ring because the == there does not conform to our standards.

**Comment by tthsqe12** [comment]
My gripes are very minor, and it would be nice to have this merged because it pulls in the tests from AA.

**Comment by fingolfin** [comment]
Sorry for the delay. I'll try to address your remarks tonight

File: test/flint/fmpq-test.jl (hunk 1 above)
Comment (fingolfin) [reply test/flint/fmpq-test.jl:3]: Thanks, I didn't know about `rand_bits`!

File: test/flint/gfp-test.jl (hunk 4 above)
Comment (fingolfin) [reply test/flint/gfp-test.jl:2]: Weird, I could have sworn I tried this and had issues. Anyway, of course that's much better, thanks!

File: test/flint/padic-test.jl (hunk 5 above)
Comment (fingolfin) [reply test/flint/padic-test.jl:10]: No worries. You already explained what went on. I just left this in as a reminder to take care of it in the future.

**Comment by fingolfin** [comment]
I've hopefully addressed all comments by @tthsqe12

**Review by wbhart — APPROVED** [review]

**Comment by tthsqe12** [comment]
looks good.
